lib/matplotlib/colors.py

```
@@ -1696,14 +1696,22 @@ def shade_rgb(self, rgb, elevation, fraction=1., blend_mode='hsv',
                 'overlay': self.blend_overlay,
                 }
         if blend_mode in lookup:
-            return lookup[blend_mode](rgb, intensity, **kwargs)
+            blend = lookup[blend_mode](rgb, intensity, **kwargs)
         else:
             try:
-                return blend_mode(rgb, intensity, **kwargs)
+                blend = blend_mode(rgb, intensity, **kwargs)
             except TypeError:
                 msg = '"blend_mode" must be callable or one of {}'
                 raise ValueError(msg.format(lookup.keys))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)